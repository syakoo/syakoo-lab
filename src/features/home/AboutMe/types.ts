export type UserLink = {
  name: string;
  url: string;
  /**
   * アイコンなどの画像の URL
   *
   * できるだけ正方形である必要がある。
   */
  imageSrc: ImageMetadata;
};

export type AboutMeConfig = {
  name: string;
  bio: string;
  imageSrc: ImageMetadata;
  links: UserLink[];
};
