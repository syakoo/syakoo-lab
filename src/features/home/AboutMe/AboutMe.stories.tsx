import type { Meta, StoryObj } from "@storybook/react";

import { AboutMe } from ".";

const meta = {
  component: AboutMe,
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta<typeof AboutMe>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {};
