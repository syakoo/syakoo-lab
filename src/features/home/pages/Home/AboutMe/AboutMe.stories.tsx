import type { Meta, StoryObj } from "@storybook/react";
import { AboutMe } from ".";

const meta = {
  component: AboutMe,
} satisfies Meta<typeof AboutMe>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {};
