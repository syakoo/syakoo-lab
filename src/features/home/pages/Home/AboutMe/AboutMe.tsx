import { aboutMeImageStyle } from "./AboutMe.css";
import { Links } from "./Links";
import { aboutMeConfig } from "./contents/aboutMeConfig";
import { Col, Row } from "@/design-system/layout";
import { Text } from "@/design-system/ui";

export const AboutMe: React.FC = () => {
  const { imageSrc, name, bio, links } = aboutMeConfig;

  return (
    <Col gap="100">
      <Row>
        <img
          alt="me"
          className={aboutMeImageStyle}
          height={imageSrc.height}
          src={imageSrc.src}
          width={imageSrc.width}
        />
        <Col gap="100">
          <Text as="h3" size="400" weight="bold">
            {name}
          </Text>
          <Links links={links} />
        </Col>
      </Row>
      <Text>{bio}</Text>
    </Col>
  );
};
