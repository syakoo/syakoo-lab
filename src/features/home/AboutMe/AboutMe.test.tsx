import { testStories } from "@/test/testStories";

import * as stories from "./AboutMe.stories";

describe("AboutMe", () => {
  testStories(stories);
});
