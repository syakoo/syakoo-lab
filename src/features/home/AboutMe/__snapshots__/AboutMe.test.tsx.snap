// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe renders Sample 1`] = `
<body>
  <div>
    <div
      class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_column__15ivi84k Flex_flexStyle_gap_200__15ivi84z Flex_flexStyle_variants_gap_200__15ivi84e Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
    >
      <div
        class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_200__15ivi84z Flex_flexStyle_variants_gap_200__15ivi84e Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
      >
        <img
          alt="me"
          class="AboutMe_aboutMeImageStyle__xung5y0"
          height="40"
          src="/img.jpg"
          width="40"
        />
        <div
          class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_column__15ivi84k Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
        >
          <h3
            class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_400__jzpfqdo Text_textStyle_variants_size_400__jzpfqd9 Text_textStyle_weight_bold__jzpfqdt"
          >
            syakoo
          </h3>
          <ul
            class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
          >
            <li>
              <a
                class="Link_linkStyle__m4v9430"
                href="https://github.com/syakoo"
              >
                <div
                  class="Links_linkStyle__dou6ce0"
                  title="Github"
                >
                  <img
                    alt="Github"
                    class="Links_linkImageStyle__dou6ce1"
                    height="40"
                    src="/img.jpg"
                    width="40"
                  />
                </div>
              </a>
            </li>
            <li>
              <a
                class="Link_linkStyle__m4v9430"
                href="https://twitter.com/sako_data"
              >
                <div
                  class="Links_linkStyle__dou6ce0"
                  title="X (Twitter)"
                >
                  <img
                    alt="X (Twitter)"
                    class="Links_linkImageStyle__dou6ce1"
                    height="40"
                    src="/img.jpg"
                    width="40"
                  />
                </div>
              </a>
            </li>
            <li>
              <a
                class="Link_linkStyle__m4v9430"
                href="https://leetcode.com/syakoo/"
              >
                <div
                  class="Links_linkStyle__dou6ce0"
                  title="LeetCode"
                >
                  <img
                    alt="LeetCode"
                    class="Links_linkImageStyle__dou6ce1"
                    height="40"
                    src="/img.jpg"
                    width="40"
                  />
                </div>
              </a>
            </li>
            <li>
              <a
                class="Link_linkStyle__m4v9430"
                href="https://steamcommunity.com/profiles/76561198944381322/"
              >
                <div
                  class="Links_linkStyle__dou6ce0"
                  title="Steam"
                >
                  <img
                    alt="Steam"
                    class="Links_linkImageStyle__dou6ce1"
                    height="40"
                    src="/img.jpg"
                    width="40"
                  />
                </div>
              </a>
            </li>
            <li>
              <a
                class="Link_linkStyle__m4v9430"
                href="https://atcoder.jp/users/syakoo"
              >
                <div
                  class="Links_linkStyle__dou6ce0"
                  title="AtCoder"
                >
                  <img
                    alt="AtCoder"
                    class="Links_linkImageStyle__dou6ce1"
                    height="40"
                    src="/img.jpg"
                    width="40"
                  />
                </div>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <p
        class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
      >
        気分駆動フロントエンドエンジニア。言語・フレームワーク仕様よりは設計やアルゴリズムなどの抽象的な部分に興味があります。
      </p>
    </div>
  </div>
</body>
`;
